### PR TITLE
devise login functioning

### PR DESCRIPTION
**Pull request by ABBennett**

### CODE REVIEW

File: spec/features/sign_in_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+require 'spec_helper'
+
+feature 'user signs up' do
+
+  scenario 'specify valid and required information' do
+    visit root_path
```
Comment (alxjrvs) [inline spec/features/sign_in_spec.rb]: Consider how using a `context` and/or `describe` block might make these feature specs DRY-er.

File: Gemfile [hunk 2]
```diff
@@ -22,6 +25,7 @@ gem 'jbuilder', '~> 2.0'
 # bundle exec rake doc:rails generates the API under doc/api.
 gem 'sdoc', '~> 0.4.0', group: :doc
 
+gem 'haml'
```
Comment (radavis) [inline Gemfile]: Looks like you have this gem listed twice in the Gemfile

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  it { should_not have_valid(:email).when('userexample.com', nil, 'users@com') }
+
+  it 'has a matching password confirmation for the password' do
+    user = User.new
+    user.password = 'password'
+    user.password_confirmation = 'anotherpassword'
+
+    expect(user).to_not be_valid
+    expect(user.errors[:password_confirmation]).to_not be_blank
+  end
```
Comment (radavis) [inline spec/models/user_spec.rb:16]: Nice unit test.

File: spec/features/sign_in_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    scenario 'password confirmation does not match confirmation' do
+      fill_in 'user_password', with: 'password'
+      fill_in 'Password Confirmation', with: 'somethingdifferent'
+
+      click_button 'Sign Up'
+      expect(page).to have_content("doesn't match")
+      expect(page).to_not have_content("Sign Out")
+    end
+  end
```
Comment (radavis) [inline spec/features/sign_in_spec.rb:38]: Nice looking feature test

File: db/migrate/20160105151541_devise_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+      t.timestamps null: false
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, :reset_password_token, unique: true
+    # add_index :users, :confirmation_token,   unique: true
+    # add_index :users, :unlock_token,         unique: true
+  end
+end
```
Comment (radavis) [inline db/migrate/20160105151541_devise_create_users.rb:43]: Hound comments on generated code can be ignored. (this was generated by `rails g devise:install`)

**Pull request merged**